PWGHF/HFJ/Tasks/taskDstarPolarisationInJet.cxx

```diff
@@ -0,0 +1,915 @@
+        if (activatePartRecoDstar && (doprocessDstarMc || doprocessDstarMcWithMl)) {
+          registry.add("hGenPartRecoPromptProduction", "THn for polarisation studies with cosThStar w.r.t. production axis -- gen prompt partly reco signal", HistType::kTHnSparseF, thnGenPromptAxes);
+          registry.add("hGenPartRecoNonPromptProduction", "THn for polarisation studies with cosThStar w.r.t. production axis -- gen non-prompt partly reco signal", HistType::kTHnSparseF, thnGenNonPromptAxes);
+        }
+      }
+      if (activateTHnSparseCosThStarJetAxis) {
+        registry.add("hGenPromptJetAxis", "THn for polarisation studies with cosThStar w.r.t. jet axis -- gen prompt signal", HistType::kTHnSparseF, thnGenPromptAxes);
+        registry.add("hGenNonPromptJetAxis", "THn for polarisation studies with cosThStar w.r.t. jet axis -- gen non-prompt signal", HistType::kTHnSparseF, thnGenNonPromptAxes);
+        if (activatePartRecoDstar && (doprocessDstarMc || doprocessDstarMcWithMl)) {
+          registry.add("hGenPartRecoPromptJetAxis", "THn for polarisation studies with cosThStar w.r.t. jet axis -- gen prompt partly reco signal", HistType::kTHnSparseF, thnGenPromptAxes);
+          registry.add("hGenPartRecoNonPromptJetAxis", "THn for polarisation studies with cosThStar w.r.t. jet axis -- gen non-prompt partly reco signal", HistType::kTHnSparseF, thnGenNonPromptAxes);
+        }
+      }
+    }
+    // inv. mass hypothesis to loop over
+    nMassHypos = 1;
+  }; // end init
+
+  /// \param invMassCharmHad is the invariant-mass of the candidate
+  /// \param ptCharmHad is the pt of the candidate
+  /// \param rapCharmHad is the rapidity of the candidate
+  /// \param invMassD0 is the invariant-mass of the D0 daugher (only for D*+)
+  /// \param cosThetaStar is the cosThetaStar of the candidate
+  /// \param outputMl is the array with ML output scores
+  /// \param isRotatedCandidate is a flag that keeps the info of the rotation of the candidate for bkg studies
+  /// \param origin is the MC origin
+  /// \param ptBhadMother is the pt of the b-hadron mother (only in case of non-prompt)
+  /// \param absEtaMin is the minimum absolute eta of the daughter tracks
+  /// \param numItsClsMin is the minimum number of ITS clusters of the daughter tracks
+  /// \param numTpcClsMin is the minimum number of TPC clusters of the daughter tracks
+  /// \param nMuons is the number of muons from daughter decays
+  /// \param isPartRecoDstar is a flag indicating if it is a partly reconstructed Dstar meson (MC only)
+  template <bool withMl, bool doMc>
+  void fillRecoHistos(charm_polarisation::CosThetaStarType cosThetaStarType, float invMassCharmHad, float ptCharmHad, float rapCharmHad, float invMassD0, float cosThetaStar, const std::array<float, 3>& outputMl, int isRotatedCandidate, int8_t origin, float ptBhadMother, float absEtaMin, int numItsClsMin, int numTpcClsMin, int8_t nMuons, bool isPartRecoDstar, float zParallel, float jetPt)
+  {
```

Comment:
The amount of parameters makes the function calls super error-prone.
It would be much safer to pass a `const&` to a single object, like a `struct` or a `map`.